front/app/components/EventCard/index.tsx

```diff
@@ -38,7 +37,7 @@ interface Props {
 const EventCard = memo<Props>((props) => {
   const { event, className, id, ...otherProps } = props;
 
-  if (!isNilOrError(event)) {
+  if (event) {
```

Comment:
I meant that the entire check can go. `event` is always defined (see its prop type). And I would also remove memo.